### ISSUE TITLE
Project Rimfactory Revived buildings missing from Industrial tab after their 2.5.0 update

### ISSUE THREAD

**Issue by joshuafcole** [DrZhivago1/DocWorld#87]
**Bug Description**

All buildings are missing from the industrial tab. 2.5.0 was a major update and defs appear to have changed.

**Screenshots**

![image](https://user-images.githubusercontent.com/313870/103328650-7ce03c80-4a0e-11eb-9c36-47eac5a831fd.png)

[**HugsLib Error Log**](https://gist.github.com/56c841efe0ce7dd7cb2f73c379f37a2a)

Nothing obviously relevant in the log, but posted in case you wanted to doublecheck!

**Aside**

Thanks for all the hard work -- this is one of the most exciting mods I've seen!

**Comment by joshuafcole** [comment]
Looks like there was a closed issue about it on the PRF: Revived GH too -- linked for posterity: https://github.com/zymex22/Project-RimFactory-Revived/issues/333

**Comment by BudickDa** [comment]
tl;dr: I think it does not longer work, because since the last update of RimFactory only PRF_BuildingBase has the designationCategory Industrial

I had the same problem. And it could be solved with the linked issue:
The Version [RimFactory Revived Legacy ](https://steamcommunity.com/sharedfiles/filedetails/?id=2340838603) works with the current Version of DocWorld. But be aware, switching between Legacy and 2.5.0 will remove most of the Rimfactory resources, research and buildings.
The buildings show up under Industrial.

But it does not work with Rimfactory Revived 2.5.0

I tried to fix it myself, by taking a look at [.Load2_DesignationCategory_Combinded.xml](https://github.com/DrZhivago1/DocWorld/blob/a44b0958897bbe4a57896809f5eed9870e01a71f/Patches/.Load2_DesignationCategory_Combined.xml#L1682) but I only found that there where two / in the path and that I have no idea how this works.

I then tried to add a category for Rimfeller, because maybe this is somehow overwriting the industrial category for RimFactory:

I added (with no sucess)
```    <Operation Class="PatchOperationFindMod">
        <mods>
            <li>Rimfeller</li>
        </mods>
        <match Class="PatchOperationSequence">
            <success>Always</success>
            <operations>
                <li Class="PatchOperationReplace">
                    <xpath>/Defs/ThingDef/designationCategory[text()="Industrial"]</xpath>
                    <value>
                        <designationCategory>DZ_Industrial</designationCategory>
                    </value>
                </li>
            </operations>
        </match>
     </Operation>
```

The designationCategory is still "Industrial" which should be replaced by DZ_Industrial by this operation:

```
                <li Class="PatchOperationReplace">
                    <xpath>/Defs/ThingDef/designationCategory[text()="Industrial"]</xpath>
                    <value>
                        <designationCategory>DZ_Industrial</designationCategory>
                    </value>
                </li>
```
My hypothesis is, that it is no longer replaced because since the update in RimFactory only PRF_BuildingBase has the designationCategory Industrial.

**Comment by joshuafcole** [comment]
@BudickDa Already got a patch in, sorry the PR didn't properly link to the issue: https://github.com/DrZhivago1/DocWorld/pull/88

The problems were:
1. When any patch fails in a PatchOperationSequence block, all latter patches are not even attempted (which would be fine, except there's no equivalent PatchOperationSet for cases where that's irrelevant :( 
2. The name of the floor signs changed and they were the first thing in the PatchOpSeq
3. Just changing the name was insufficient to fix it, because they no longer had a designationCategory tag (instead inheriting it from a parent) so PatchOperationReplace had to be switched with PatchOperationAdd
4. No diagnostic information in the log about _why_ the patches failed in any of the permutations I tried until I got it right. :P 

The patching situation for RW is not super great to debug atm. :(

**Comment by DrZhivago1** [comment]
I'll review this soon, thanks for the info and thoughts lads
